### PR TITLE
OSX-580 Reporting - email and PDF functionality (2.2.0-alpha2)

### PR DESCRIPTION
**Pull request by ifarzana**

### CODE REVIEW

File: views/settings-page.twig [hunk 1]
```diff
@@ -401,6 +401,17 @@
                                     {% endif %}
                                 {% endif %}
 
+                                {% if theme.isSettingVisible("QUICK_CHART_URL") %}
+                                    {% set title %}{% trans "Quick Chart URL" %}{% endset %}
+                                    {% set helpText %}{% trans "Quick Chart URL for reporting and 2FA." %}{% endset %}
```
Comment (dasgarner) [inline views/settings-page.twig]: Perhaps:

"Enter the URL to a Quick Chart service. This is used to draw charts in emailed reports and for showing a QR code during two factor authentication."

File: docker-compose.yml [hunk 2]
```diff
@@ -39,5 +39,10 @@ services:
     environment:
     - API_URL=http://localhost/swagger.json
 
+  quickchart:
+    image: ianw/quickchart
+    ports:
+    - "3400:3400"
```
[the remark this answers is not in the document]
Comment (ifarzana) [reply docker-compose.yml]: Shall I remove the ports from here?

File: lib/XTR/NotificationTidyTask.php [hunk 3]
```diff
@@ -85,6 +85,21 @@ public function run()
...
+        $sql = 'SELECT filename  FROM `notification` ' . $where;
+
+        foreach ($this->store->select($sql, $params) as $row) {
+            $filename = $row['filename'];
+
+            /*Delete the attachment*/
+            if (!empty($filename)) {
+                // Library location
+                $attachmentLocation = $this->config->getSetting('LIBRARY_LOCATION'). 'attachment/';
+                if (file_exists($attachmentLocation . $filename))
```
Comment (dasgarner) [inline lib/XTR/NotificationTidyTask.php]: `{}`

File: lib/XTR/ReportScheduleTask.php [hunk 4]
```diff
@@ -108,10 +108,10 @@ private function runReportSchedule()
             if ($nextRunDt <= $now) {
 
                 // random run of report schedules
-                $skip = $this->skipReportRun($now, $nextRunDt);
-                if ($skip == true) {
-                    continue;
-                }
+//                $skip = $this->skipReportRun($now, $nextRunDt);
```
Comment (dasgarner) [inline lib/XTR/ReportScheduleTask.php]: `//` ?

File: lib/XTR/ReportScheduleTask.php [hunk 5]
```diff
@@ -211,38 +216,45 @@ private function createPdfAndNotification($reportSchedule, $savedReport, $media)
...
+                $notification->releaseDt = time() + 15 * 60 ; // 15 minutes after the notification is created so that the notification task can pick this
+                $notification->isEmail = 1;
+                $notification->isInterrupt = 0;
+                $notification->userId = $savedReport->userId; // event owner
+                $notification->filename = 'filename-'.$media->mediaId.'.pdf';
+                $notification->nonusers = $nonusers;
+                $notification->save();
+            }
+        } catch (\Exception $error) {
+            $this->log->error('Report PDF could not be created and notifiction is not saved.');
```
Comment (dasgarner) [inline lib/XTR/ReportScheduleTask.php]: would it be useful to add this to the task status message?

**Pull request merged**